modules/crafted-startup-config.el

```diff
@@ -88,30 +102,76 @@ Each element in the list should be a list of strings or pairs
                 (custom-save-all))
               (quit-windows-on "*Crafted Emacs*" t)))
    "  ")
-   (when custom-file
-     (let ((checked (create-image "checked.xpm"
-                                  nil nil :ascent 'center))
-           (unchecked (create-image "unchecked.xpm"
-                                    nil nil :ascent 'center)))
-       (insert-button
-        " "
-        :on-glyph checked
-        :off-glyph unchecked
-        'checked nil 'display unchecked 'follow-link t
-        'action (lambda (button)
-                  (if (overlay-get button 'checked)
-                      (progn (overlay-put button 'checked nil)
-                             (overlay-put button 'display
-                                          (overlay-get button :off-glyph))
-                             (setq crafted-startup-screen-inhibit-startup-screen
-                                   nil))
-                    (overlay-put button 'checked t)
-                    (overlay-put button 'display
-                                 (overlay-get button :on-glyph))
-                    (setq crafted-startup-screen-inhibit-startup-screen t))))))
+  (when custom-file
+    (let ((checked (create-image "checked.xpm"
+                                 nil nil :ascent 'center))
+          (unchecked (create-image "unchecked.xpm"
+                                   nil nil :ascent 'center)))
+      (insert-button
+       " "
+       :on-glyph checked
+       :off-glyph unchecked
+       'checked nil 'display unchecked 'follow-link t
+       'action (lambda (button)
+                 (if (overlay-get button 'checked)
+                     (progn (overlay-put button 'checked nil)
+                            (overlay-put button 'display
+                                         (overlay-get button :off-glyph))
+                            (setq crafted-startup-screen-inhibit-startup-screen
+                                  nil))
+                   (overlay-put button 'checked t)
+                   (overlay-put button 'display
+                                (overlay-get button :on-glyph))
+                   (setq crafted-startup-screen-inhibit-startup-screen t))))))
   (fancy-splash-insert :face '(variable-pitch (:height 0.9))
                        " Never show it again."))
 
+(defun crafted-startup-diary ()
+  (require 'diary-lib nil :noerror)
+  (when (diary-check-diary-file)
```

Comment:
Not sure it's intended, but this errors when there is no diary file.
This results in the entire startup screen not loading.

Given it's used as a guard here, I think `(file-readable-p diary-file)` makes more sense?

Reply:
Shucks... missed that one.  Yes, the `diary-check-diary-file` throws errors, I was just trying not to re-implement that function, but I guess I need to.  Thanks for point it out.